src/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -30,6 +30,22 @@ public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source)
             return source is IIListProvider<TSource> listProvider ? listProvider.ToList() : new List<TSource>(source);
         }
 
+        public static List<TSource> ToList<TSource>(this IEnumerable<TSource> source, int capacity)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), "Non-negative number required.");
+            }
+
+            var list = new List<TSource>(capacity);
```

Comment:
What if the calling code passes a value lower than the current `ToList()` would have calculated for the capacity? This would cause more resizes then, as well as yet more when the caller adds in more elements (since if they aren't planning to add a known number of elements there isn't much to gain here, so this is the only real use-case). Should we trust the caller to always be correct (if they aren't it won't actually break) or should we override with the existing logic if it comes up with a higher number?